test/unit/spec/IDPDiscovery_spec.js

```diff
@@ -0,0 +1,1031 @@
+        });
+      });
+      itp('does not re-render rememberMe checkbox on changes', function () {
+        Util.mockCookie('ln', 'testuser@clouditude.net');
+        var options = {
+          'features.rememberMe': true
+        };
+        return setup(options).then(function (test) {
+          var orig = test.form.rememberMeCheckbox().get(0);
+          test.form.setUsername('new-user@clouditude.net');
+          expect(test.form.rememberMeCheckbox().get(0)).toBe(orig);
+        });
+      });
+      itp('populates username if username is available', function () {
+        var options = {
+          'username': 'testuser@ABC.com'
+        };
+        return setup(options).then(function (test) {
+          expect(test.form.usernameField().val()).toBe('testuser@ABC.com');
+        });
+      });
+      itp('populates username if username is available and when features.rememberMe is false', function () {
+        var options = {
+          'features.rememberMe': false,
+          'username': 'testuser@ABC.com'
+        };
+        return setup(options).then(function (test) {
+          var cb = test.form.rememberMeCheckbox();
+          expect(cb.length).toBe(0);
+          expect(test.form.usernameField().val()).toBe('testuser@ABC.com');
+        });
+      });
+      itp('ignores lastUsername and hides rememberMe if features.rememberMe is false and cookie is set', function () {
+        Util.mockCookie('ln', 'testuser@ABC.com');
+        var options = {
+          'features.rememberMe': false
+        };
+        return setup(options).then(function (test) {
+          var cb = test.form.rememberMeCheckbox();
+          expect(cb.length).toBe(0);
+          expect(test.form.usernameField().val().length).toBe(0);
+        });
+      });
+      itp('unchecks rememberMe if username is populated and lastUsername is different from username', function () {
+        Util.mockCookie('ln', 'testuser@clouditude.net');
+        var options = {
+          'features.rememberMe': true,
+          'username': 'testuser@ABC.com'
+        };
+        return setup(options).then(function (test) {
+          expect(test.form.rememberMeCheckboxStatus()).toBe('unchecked');
+          expect(test.form.usernameField().val()).toBe('testuser@ABC.com');
+        });
+      });
+      itp('checks rememberMe if username is populated and lastUsername is same as username', function () {
+        Util.mockCookie('ln', 'testuser@ABC.com');
+        var options = {
+          'features.rememberMe': true,
+          'username': 'testuser@ABC.com'
+        };
+        return setup(options).then(function (test) {
+          expect(test.form.rememberMeCheckboxStatus()).toBe('checked');
+          expect(test.form.usernameField().val()).toBe('testuser@ABC.com');
+        });
+      });
+      itp('shows an error if username is empty and submitted', function () {
+        return setup().then(function (test) {
+          test.setNextWebfingerResponse({});
```

Comment:
we probably don't need to do this, there is a validation error before we submit the form

Reply:
This is testing that the webfinger call is never made. It expects a spy for that reason.